internal/services/mongocluster/mongo_cluster_resource.go

```diff
@@ -393,6 +525,7 @@ func (r MongoClusterResource) Update() sdk.ResourceFunc {
 			if metadata.ResourceData.HasChange("storage_size_in_gb") {
 				payload.Properties.Storage = &mongoclusters.StorageProperties{
 					SizeGb: pointer.To(state.StorageSizeInGb),
+					Type:   pointer.To(mongoclusters.StorageType(state.StorageType)),
```

Comment:
`pointer.ToEnum` is preferred when working with enums.
```suggestion
					Type:   pointer.ToEnum[mongoclusters.StorageType](state.StorageType),
```